pinot-controller/src/main/java/com/linkedin/pinot/controller/helix/core/PinotHelixResourceManager.java

```diff
@@ -353,33 +355,34 @@ public synchronized PinotResourceManagerResponse addInstance(Instance instance)
   }
 
   /**
-   * TODO : Due to the issue of helix, if remove the segment from idealState directly, the node won't get transaction,
-   * so we first disable the segment, then remove it from idealState, then enable the segment. This will trigger transactions.
-   * After the helix patch, we can refine the logic here by directly drop the segment from idealState.
-   *
-   * @param tableName
-   * @param segmentId
+   * Deletes segment from helix idealstate and drop it from local storage
+   * @param tableName Segment's table name
+   * @param segmentName segment name
    * @return
    */
-  public synchronized PinotResourceManagerResponse deleteSegment(final String tableName, final String segmentId) {
-    LOGGER.info("Trying to delete segment: {} for table: {} ", segmentId, tableName);
+  public synchronized PinotResourceManagerResponse deleteSegment(final String tableName, final String segmentName) {
+    return deleteSegments(tableName, Arrays.asList(segmentName));
+  }
+
+  /**
+   * Deletes a list of segments from idealstate and drop those from the local storage
+   * @param tableName Name of segment's table
+   * @param segments List of segment names
+   * @return
+   */
+  public synchronized PinotResourceManagerResponse deleteSegments(final String tableName, final List<String> segments) {
+    LOGGER.info("Trying to delete segments: {} for table: {} ", StringUtils.join(segments, ','), tableName);
     final PinotResourceManagerResponse res = new PinotResourceManagerResponse();
     try {
-      IdealState idealState = _helixAdmin.getResourceIdealState(_helixClusterName, tableName);
-      if (idealState == null || idealState.getPartitionSet() == null || !idealState.getPartitionSet().contains(segmentId)) {
-        res.message = "Segment " + segmentId + " not in IDEALSTATE.";
-        LOGGER.info("Segment: {} is not in IDEALSTATE", segmentId);
-      } else {
-        LOGGER.info("Trying to delete segment: {} from IdealStates", segmentId);
-        HelixHelper.removeSegmentFromIdealState(_helixZkManager, tableName, segmentId);
-        res.message = "";
+      HelixHelper.removeSegmentsFromIdealState(_helixZkManager, tableName, segments);
+      for (String segment : segments) {
+        _segmentDeletionManager.deleteSegment(tableName, segment);
       }
-      _segmentDeletionManager.deleteSegment(tableName, segmentId);
 
-      res.message += "Segment " + segmentId + " successfully deleted.";
+      res.message += "Segment " + StringUtils.join(segments, ',') + " successfully deleted.";
```

Comment:
Remove this log statement. SegmentDeletionManager already logs. 
A useful log here may be how many segments are deleted. 
In any case, let us not have all the segment names appeneded, since the string can get arbitrarily long